### ISSUE TITLE
Clarification: How much of ‘base uri’ should be honored?

### ISSUE THREAD

**Issue by steveluscher** [solana-mobile/mobile-wallet-adapter#58]
Regarding the ‘[Endpoint-specific URIs](https://github.com/solana-labs/mirror-mobile-wallet-adapter/blob/main/spec/spec.md#endpoint-specific-uris)’ section of the spec.

* Should the _scheme_ that the wallet responds with be honored, or should we always use `https`?
* Should the _query_ parameters the wallet responds with be honored, or should we strip them?
* Should the _fragment_ the wallet responds with be honored, or should we strip it?

In other words, if the association path for the protocol is `/v1/associate/local?association=<association_token>&port=<port_number>` what URL should we construct if the wallet responds with:

1. `instagram://wallet`
2. `https://mydapp.com/mobilewallet/?utm_source=foo`
3. `https://mydapp.com/mobilewallet/?utm_source=foo&port=4321`
4. `https://mydapp.com/mobilewallet/#user=abc`
